src/optimizer/plan_generator.cpp

```diff
@@ -133,9 +133,34 @@ void PlanGenerator::Visit(const QueryDerivedScan *) {
 }
 
 void PlanGenerator::Visit(const PhysicalLimit *op) {
+  // Generate order by plan when there's internal sort order
+  if (!op->sort_exprs.empty()) {
```

Comment:
So `sort_exps` is not empty means there's no internal sort order?

Reply:
`sort_exprs` is not empty means there is internal sort order in this case.